pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -168,29 +368,91 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		// Make a deep copy so we don't mutate the shared cache
 		copy, err := scheme.Scheme.DeepCopy(replicas[i])
 		if err != nil {
-			return err
+			return &status, err
 		}
 		replica := copy.(*v1.Pod)
-		if err := ssc.podControl.UpdateStatefulPod(set, replica); err != nil {
-			return err
+		if err := ssc.podControl.UpdateStatefulPod(updateSet, replica); err != nil {
+			return &status, err
 		}
 	}
 
 	// At this point, all of the current Replicas are Running and Ready, we can consider termination.
 	// We will wait for all predecessors to be Running and Ready prior to attempting a deletion.
 	// We will terminate Pods in a monotonically decreasing order over [len(pods),set.Spec.Replicas).
-	// Note that we do not resurrect Pods in this interval.
+	// Note that we do not resurrect Pods in this interval. Also not that scaling will take precedence over
+	// updates.
 	if unhealthy > 0 && monotonic {
 		glog.V(2).Infof("StatefulSet %s is waiting on %d Pods", set.Name, unhealthy)
-		return nil
+		return &status, nil
 	}
 	for target := len(condemned) - 1; target >= 0; target-- {
 		glog.V(2).Infof("StatefulSet %s terminating Pod %s", set.Name, condemned[target])
 		if err := ssc.podControl.DeleteStatefulPod(set, condemned[target]); err != nil {
-			return err
+			return &status, err
+		}
+		if getPodRevision(pods[target]) == currentRevision.Name {
+			status.CurrentReplicas--
+		} else if getPodRevision(pods[target]) == updateRevision.Name {
+			status.UpdatedReplicas--
 		}
 		if monotonic {
-			return nil
+			return &status, nil
+		}
+	}
+
+	// for the OnDelete strategy we short circuit. Pods will be updated when they are manually deleted.
+	if set.Spec.UpdateStrategy.Type == apps.OnDeleteStatefulSetStrategyType {
+		return &status, nil
+	}
+
+	// we compute the minimum ordinal of the target sequence for a destructive update based on the strategy.
+	updateMin := 0
+	if set.Spec.UpdateStrategy.Type == apps.PartitionStatefulSetStrategyType {
```

Comment:
I see Partitioned handled only for deletion but not for updation, probably i am not understanding it @kow3ns